### PR TITLE
Some tidying on the code

### PR DESCRIPTION
**Pull request by akirak**
I made some tidying on the code while retaining the current behavior:

- Use `version<=` for comparing two version strings. Alternatively, you can use `version-to-list` to parse the version string.
- Use `rx` macro to match against the version string. `rx` is convenient in many situations.
- Use `when` when there is no need to handle the alternative case. `when` is better because you can sequence the inner expressions without using `progn`.
- You don't need `progn` for evaluating multiple expressions in the "else" case inside `if`.
- Replace `let*` with `let` where it's sufficient.

### CODE REVIEW

**Comment by djgoku** [comment]
Thank you! It might take me a few days to get this reviewed. I look forward to check out the changes!

File: sops.el [hunk 1]
```diff
@@ -65,8 +65,8 @@
   :group 'sops
   :lighter " sops"
 
-  (if (eq sops--sops-version-greater-than-equal-to-3-9 -1)
-      (sops--version-check))
+  (when (= sops--sops-version-greater-than-equal-to-3-9 -1)
```
Comment (djgoku) [inline sops.el]: `(when (eq ... -1)` fails when `sops--sops-version-greater-than-equal-to-3-9` is `t` or `nil`.

We should change it back to `eq` for now.

File: sops.el [hunk 2]
```diff
@@ -94,109 +94,95 @@
...
-                  (setq sops--sops-version-greater-than-equal-to-3-9 t)
-                  t)
-              (progn
-                (setq sops--sops-version-greater-than-equal-to-3-9 nil)
-                nil)))))))
+    (when (zerop (call-process sops-executable nil t nil "--version"))
+      (goto-char (point-min))
+      (re-search-forward (rx (+ digit) (+ (and "." (+ digit)))))
+      (setq sops--sops-version-greater-than-equal-to-3-9
+            (version<= "3.9" (match-string 0))))))
```
Comment (djgoku) [inline sops.el]: As I was reviewing I couldn't figure out how `(version<= "3.9" (match-string 0))` actually worked. But I would another solution I think I like better `(not (string-version-lessp (match-string 0) "3.9.0"))`

Edit 1:

I messed up and changed `version<=` before I looked up. Actually this is correct.

Edit 2:

Actually we want:

`(not (version< (match-string 0) "3.9"))`

**Review by djgoku — CHANGES_REQUESTED** [review]
I appreciate your changes and tests are still passing with your changes and the changes I brought up in the review. I'll let you make the changes and review them once the changes are pushed.

Also, I learned a few new things from your changes!

Sorry, it took so long to get back to you.

File: sops.el (hunk 2 above)
Comment (djgoku) [reply sops.el:185]: I messed up and changed `version<=` before I looked up. Actually this is correct.

Comment (akirak) [reply sops.el]: Well, `(match-string 0)` is the sops version, so `(version<= "3.9" (match-string 0))` means the sops version `>=` (is greater than or equal to) 3.9, doesn't it? However, your suggestion is totally fine, so I will follow your preference.

File: sops.el (hunk 1 above)
Comment (akirak) [reply sops.el]: I see. You are right. `=` only supports numeric comparison.

**Comment by akirak** [comment]
I have fixed the issues.

I would also suggest addition of either [eask](https://github.com/emacs-eask/cli) or [makem](https://github.com/alphapapa/makem.sh/) for running checks on CI.

**Pull request merged**